architecture/backend_upstream.md

```diff
@@ -0,0 +1,24 @@
+# Backend and Upstream Testing
+
+Sometimes you need to debug an issue on production without varnish being in your way.
```

Comment:
Added a bit of text to reflect the behaviour of varnish on development and production :)